### ISSUE TITLE
[ERR_PACKAGE_PATH_NOT_EXPORTED] estree- walker/package.json

### ISSUE THREAD

**Issue by boydaihungst** [unjs/unhead#119]
### Environment

Linux: EndeavourOS Linux x86_64
Kernel: 6.1.12-arch1-1

Node: v18.13.0
@unhead/addons: 1.1.5
@unhead/vue: 1.0.21
yarn: 1.22.19

### Reproduction

1. Update to unhead packages to 1.1.5 
2. Run: yarn vite

### Describe the bug

After the [1.1.0](https://github.com/unjs/unhead/releases/tag/v1.1.0) release just got that error.

### Additional context

My vite.config.ts file
```ts
/* eslint-env node */
import VueI18nPlugin from '@intlify/unplugin-vue-i18n/vite';
import UnheadVite from '@unhead/addons/vite';
import LegacyPlugin from '@vitejs/plugin-legacy';
import vue from '@vitejs/plugin-vue';
import vueJsx from '@vitejs/plugin-vue-jsx';
import path from 'node:path';
import { fileURLToPath, URL } from 'node:url';
import { loadEnv, type BuildOptions } from 'vite';
import { compression } from 'vite-plugin-compression2';
import tsconfigPaths from 'vite-tsconfig-paths';
import { coverageConfigDefaults, defineConfig } from 'vitest/config';
import * as zlib from 'zlib';

// https://vitejs.dev/config/

/**
 * Build optimized for production
 */
const buildProd: BuildOptions = {
  rollupOptions: {
    // exclude mock server worker
    external: [
      fileURLToPath(new URL('public/mockServiceWorker.js', import.meta.url)),
      fileURLToPath(new URL('tests/mock-api/*', import.meta.url)),
    ],
    plugins: [
      // TsconfigPathsPlugin({}),
      UnheadVite(),
      compression({
        algorithm: 'gzip',
        exclude: [/\.(br)$/, /\.(gz)$/],
        include: [/\.js$|\.css$|\.eot?.+$|\.ttf?.+$|\.woff?.+$|\.svg?.+$/],
        deleteOriginalAssets: false,
        threshold: 0,
      }),
      compression({
        algorithm: 'brotliCompress',
        exclude: [/\.(br)$/, /\.(gz)$/],
        include: [/\.js$|\.css$|\.eot?.+$|\.ttf?.+$|\.woff?.+$|\.svg?.+$/],
        deleteOriginalAssets: false,
        threshold: 0,
        compressionOptions: {
          params: {
            [zlib.constants.BROTLI_PARAM_QUALITY]: 11,
          },
        },
      }),
    ],
  },
};

/**
 * Strip data-test from vue template
 */
function removeAttrFromTemplate(node: any, attrName: string) {
  if (node.type === 1 /* NodeTypes.ELEMENT */) {
    node.props = node.props.filter((prop: any) =>
      prop.type === 6 /* NodeTypes.ATTRIBUTE */ ? prop.name !== attrName : true,
    );
  }
}
export default defineConfig(({ mode }) => {
  // Default vite won't load env in config vite config file
  Object.assign(process.env, loadEnv(mode, process.cwd(), ''));
  const isProd = mode === 'production';
  const pathMap: Record<string, string> = {
    // '@tests': './tests',
    // '@src': './src',
    // '@router': './src/router',
    // '@views': './src/router/views',
    // '@layouts': './src/router/layouts',
    // '@components': './src/components',
    // '@assets': './src/assets',
    // '@utils': './src/utils',
    // '@stores': './src/stores',
    // Every @import inside style tag need to be defined here. Until [issue-1052536338](https://github.com/aleclarson/vite-tsconfig-paths/issues/41#issue-1052536338) is solved
    '@design': './src/design/index.scss',
    // // custom
    // '@plugin': './src/plugin',
    // '@models': './src/models',
    // '@composables': './src/composables',
  };
  const resolveAlias: Record<string, any> = {};

  for (const alias in pathMap) {
    const aliasTo = pathMap[alias];

    resolveAlias[alias] = path.resolve(__dirname, aliasTo);
    // fileURLToPath(new URL(aliasTo, import.meta.url));
  }
  return {
    // single page app mode
    appType: 'spa',
    // server when run vite dev
    server: {
      /**
       * We use msw's web service worker to mock server response locally
       * {@link https://mswjs.io/docs/}
       */
      host: 'localhost',
      // set to true to automatically open in browser on startup
      open: false,
      port: 8081,
      /** {@link https://github.com/chimurai/http-proxy-middleware} */
      proxy: {
        '/api': {
          // real api url to test or run app, only forward request to mock server if this env variable is not existed
          target: process.env.VITE_API_BASE_URL,
          secure: false,
          ws: true,
          changeOrigin: true,
        },
      },
    },
    // {@link https://vitejs.dev/config/shared-options.html#define}
    define: {
      'import.meta.env.PROD': isProd,
      // {@link https://vitest.dev/config/#configuration}
      'import.meta.vitest': process.env.VITE_TEST,
      ...(process.env.VITE_TEST
        ? {
            // global function for testing unit only
            mount: null,
            shallowMount: null,
            shallowMountView: null,
            createComponentMocks: null,
            createModuleStore: null,
          }
        : {}),
    },
    build: isProd ? buildProd : {},
    // settings for vitest
    test: {
      globals: true,
      environment: 'jsdom',
      environmentOptions: {},
      setupFiles: [
        './tests/unit/setup.ts',
        './tests/unit/setup-mock-server.ts',
        './tests/unit/matchers.ts',
      ],
      globalSetup: ['./tests/unit/global-setup.ts'],
      // extras alias that will be added to resolve.alias below
      alias: {},
      coverage: {
        reporter: ['text', 'json', 'html'],
        reportsDirectory: './tests/unit/coverage',
        include: ['src/**/*.{js,vue,ts,jsx,tsx,cjs,mjs,cts,mts}'],
        exclude: [
          ...coverageConfigDefaults.exclude,
          'src/**/_globals.ts',
          'src/**/*.d.ts',
          '**/node_modules/**',
          'src/main.ts',
          'src/App.vue',
          'src/router/index.ts',
          'src/router/routes.ts',
          'src/stores/store.ts',
          'src/stores/helpers.ts',
          'src/stores/index.ts',
        ],
        src: ['./src'],
      },
      // For mocking css module
      css: {
        modules: {
          classNameStrategy: 'non-scoped',
        },
      },
    },
    plugins: [
      // auto add resolve.alias from tsconfig files
      tsconfigPaths({
        loose: true,
      }),
      vue({
        isProduction: isProd,
        template: {
          compilerOptions: {
            // Remove any `data-test` prop, attribute from vue templates
            nodeTransforms: isProd
              ? [(node: any) => removeAttrFromTemplate(node, 'data-test')]
              : [],
          },
        },
      }),
      // support jsx
      vueJsx(),
      VueI18nPlugin({
        runtimeOnly: isProd,
        compositionOnly: true,
        fullInstall: false,
      }),
      // build for legacy browser
      LegacyPlugin({
        targets: ['defaults', 'not IE 11'],
      }),
    ] as any,
    resolve: {
      alias: {
        ...resolveAlias,
      },
    },
    css: {
      devSourcemap: !isProd,
    },
  };
});

```

### Logs

```sh
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: No "exports" main defined in /home/huyhoang/git/vue-default/node_modules/@unhead/addons/node_modules/estree-
walker/package.json
    at new NodeError (node:internal/errors:400:5)
    at exportsNotFound (node:internal/modules/esm/resolve:361:10)
    at packageExportsResolve (node:internal/modules/esm/resolve:641:13)
    at resolveExports (node:internal/modules/cjs/loader:538:36)
    at Module._findPath (node:internal/modules/cjs/loader:607:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:1025:27)
    at Module._load (node:internal/modules/cjs/loader:885:27)
    at Module.require (node:internal/modules/cjs/loader:1105:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (/home/huyhoang/git/vue-default/node_modules/@unhead/addons/dist/shared/addons.1b8e1760.cjs:10:22) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```

**Comment by harlan-zw** [comment]
Hey @boydaihungst 

Not sure what the issue here is, think it might be a node version problem, I'm using the 16 LTS if it helps.

Regardless, I've pushed up a commit that inlines the `estree-walker` dependency which should solve this.

Available in 1.1.6

Let me know how you go.

**Comment by boydaihungst** [comment]
It's fixed. Thank you so much.
